tests/e2e/tests/pilot/grpc_test.go

```diff
@@ -16,75 +16,55 @@ package pilot
 
 import (
 	"fmt"
-
-	meshconfig "istio.io/api/mesh/v1alpha1"
-	tutil "istio.io/istio/tests/e2e/tests/pilot/util"
+	"testing"
 )
 
-type grpc struct {
-	*tutil.Environment
-	logs *accessLogs
-}
-
-func (t *grpc) String() string {
-	return "http2-reachability"
-}
-
-func (t *grpc) Setup() error {
-	t.logs = makeAccessLogs()
-	return nil
-}
-
-func (t *grpc) Teardown() {
-}
-
-func (t *grpc) Run() error {
-	if err := t.makeRequests(); err != nil {
-		return err
-	}
-	return t.logs.check(t.Environment)
-}
-
-func (t *grpc) makeRequests() error {
-	// Auth is enabled for d:7070 using per-service policy. We expect request
-	// from non-envoy client ("t") should fail all the time.
+func TestGrpc(t *testing.T) {
+	// Auth is enabled for d:80, and disabled for d:8080 using per-service policy.
+	// We expect request from non-envoy client ("t") to d:80 should always fail,
+	// while to d:8080 should always success.
 	srcPods := []string{"a", "b"}
 	dstPods := []string{"a", "b", "d"}
-	if t.Auth == meshconfig.MeshConfig_NONE {
+	ports := []string{":70", ":7070"}
+	if !tc.Kube.AuthEnabled {
 		// t is not behind proxy, so it cannot talk in Istio auth.
 		srcPods = append(srcPods, "t")
 		dstPods = append(dstPods, "t")
 		// mTLS is not supported for headless services
 		dstPods = append(dstPods, "headless")
 	}
-	funcs := make(map[string]func() tutil.Status)
-	for _, src := range srcPods {
-		for _, dst := range dstPods {
-			for _, port := range []string{":70", ":7070"} {
-				for _, domain := range []string{"", "." + t.Config.Namespace} {
-					name := fmt.Sprintf("GRPC request from %s to %s%s%s", src, dst, domain, port)
-					funcs[name] = (func(src, dst, port, domain string) func() tutil.Status {
-						url := fmt.Sprintf("grpc://%s%s%s", dst, domain, port)
-						return func() tutil.Status {
-							resp := t.ClientRequest(src, url, 1, "")
+
+	logs := newAccessLogs()
+
+	// Run all request tests.
+	t.Run("request", func(t *testing.T) {
```

Comment:
grpc?

Reply:
The test is already "TestGrpc". This will appear as "TestGrpc/request".